spec/models/bishop_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Bishop, type: :model do
+  describe '#move_legal?' do
+    let(:bishop){Bishop.create(row: 3, col: 3)}
+
+    it "is legal" do
+      expect(bishop.move_legal?(4,4)).to be true

```

Comment:
I may just be nitpicking here. My point was that the test you wrote would also work for a king piece (it can move diagonally by one row and col). Using a square like (6,6) shows that your code allows a bishop to move diagonally by more than just one row and col.